### PR TITLE
Distribution, expand the ignored distributios set

### PR DESCRIPTION
**Pull request by ValarDragon**
Expands the set of gauges we ignore, as we are now no longer concerned about state compat in main.

This changes
```
if remainCoins.Len() == 1 && remainCoins[0].Amount.LTE(osmomath.NewInt(10)) && gauge.DistributeTo.Denom == "gamm/pool/1" && remainCoins[0].Denom != "uosmo" {
```
to 
```
	// Remove some spam gauges that are not worth distributing. (We ignore the denom stake because of tests.)
	if remainCoins.Len() == 1 && remainCoins[0].Amount.LTE(osmomath.NewInt(100)) && remainCoins[0].Denom != "stake" {
		ctx.Logger().Debug(fmt.Sprintf("gauge debug, this gauge is perceived spam, skipping %d", gauge.Id))
		err := k.updateGaugePostDistribute(ctx, gauge, totalDistrCoins)
		return true, totalDistrCoins, err
	}
	return false, totalDistrCoins, nil
}
```

Requiring a gauge to distribute at least 100 coins, for any pool. Ignores "stake" to avoid fixing all the tests.

### CODE REVIEW

File: x/incentives/keeper/distribute.go [hunk 1]
```diff
@@ -713,6 +698,28 @@ func (k Keeper) distributeInternal(
...
+
+	// In this case, remove redundant cases.
+	// Namely: gauge empty OR gauge coins undistributable.
+	if remainCoins.Empty() {
+		ctx.Logger().Debug(fmt.Sprintf("gauge debug, this gauge is empty, why is it being ran %d. Balancer code", gauge.Id))
+		err := k.updateGaugePostDistribute(ctx, gauge, totalDistrCoins)
+		return true, totalDistrCoins, err
+	}
+
+	// Remove some spam gauges that are not worth distributing. (We ignore the denom stake because of tests.)
```
Comment (czarcas7ic) [inline x/incentives/keeper/distribute.go:714]: > (We ignore the denom stake because of tests.)

LOL, cant we just inc the tests to be greater than 100??

**Review by czarcas7ic — APPROVED** [review]

File: x/incentives/keeper/distribute.go (hunk 1 above)
Comment (ValarDragon) [reply x/incentives/keeper/distribute.go:714]: yeah but its pain

File: x/incentives/keeper/distribute.go [hunk 2]
```diff
@@ -713,6 +698,28 @@ func (k Keeper) distributeInternal(
 	return totalDistrCoins, err
 }
 
+func (k Keeper) skipSpamGaugeDistribute(ctx sdk.Context, locks []*lockuptypes.PeriodLock, gauge types.Gauge, totalDistrCoins sdk.Coins, remainCoins sdk.Coins) (bool, sdk.Coins, error) {
```
Comment (mattverse) [inline x/incentives/keeper/distribute.go:701]: I think it'd be worth creating unit test for this as well

**Review by mattverse — APPROVED** [review]

File: x/incentives/keeper/distribute.go (hunk 2 above)
Comment (czarcas7ic) [reply x/incentives/keeper/distribute.go:701]: Added https://github.com/osmosis-labs/osmosis/pull/7250/commits/5cf383a0df80454fad9f4c202f02dc903c9fef3b

**Pull request merged**